build.gradle

```diff
@@ -76,4 +76,132 @@ project(':configurator') {
                     'Main-Class': mainClassName)
         }
     }
-}
\ No newline at end of file
+}
+ext {
+    osNames = ['linux', 'windows', 'mac']
+    jreFileNameTemplate = {sys,extension -> "OpenJDK8U-jre_x64_${sys}_hotspot_8u212b03.${extension}" }
+    jreDownloadDir = file("$buildDir/jre-downloads")
+    jreExtractDir = "${jreDownloadDir.absolutePath}/jre"
+    jreExtractSubDir = {sys -> "${jreExtractDir}/$sys"}
+    jreHashes = ['74daf0b77a7fd679cbb3a6228e0efa8c4a90b7664aa057f211e34bbfb38640fb',
+                 'c09bab89cd82483c371597c5c364094a145c1fbba43a1d3d7c3e350b89dedc89',
+                 '80855320c42a3b06617c6e466c64df67731542a805972185b075ca6cb1222c7f']
+    packageFileExtension = ['zip', 'tar.gz']
```

Comment:
This makes for some rather unreadable and error-prone code below. I'd suggest either having a map or enum, or seeing they're just two values how about two constants?

The same goes for `osNames` above. Although I understand with that one you wish to iterate across, but all the indices down in `setPackageProperties` is not the most readable to be sure.